forecast_tools.R

```diff
@@ -20,12 +20,42 @@ FullPath <- function( ReferencePath, BasePath=getwd()){
   return (Path)
 }
 
+#Get all model aic values and calculate akaike weights
+compile_aic_weights = function(forecast_folder='./predictions'){
+  model_aic_filenames = list.files(forecast_folder, full.names = TRUE, recursive = TRUE)
+  model_aic_filenames = model_aic_filenames[grepl('aic_weights',model_aic_filenames)]
+
+  all_model_aic = purrr::map(model_aic_filenames, ~read.csv(.x, na.strings = '', stringsAsFactors = FALSE)) %>% 
+    bind_rows()
+
+  all_weights = all_model_aic %>%
+    group_by(date,currency, level, species) %>%
+    mutate(delta_aic = aic-min(aic), weight = exp(-0.5*delta_aic) / sum(exp(-0.5*delta_aic))) %>%
+    ungroup()
+  return(all_weights)
+}
+
 #Create the ensemble model from all other forecasts
-#Currently just the mean of the esimates and confidence intervals.
-make_ensemble=function(all_forecasts, model_weights=NA, models_to_use=NA){
-  ensemble = all_forecasts %>%
+#Uses the weighted mean and weighted sample variance
+#https://en.wikipedia.org/wiki/Weighted_arithmetic_mean
+make_ensemble=function(all_forecasts, models_to_use=NA, CI_level = 0.9){
+  weights = compile_aic_weights()
+  
+  CI_sd = qnorm((1-CI_level)/2, lower.tail = FALSE)
+  
+  weighted_estimates = all_forecasts %>%
+    left_join(weights, by=c('date','model','currency','level','species')) %>%
     group_by(date, NewMoonNumber, forecastmonth, forecastyear,level, currency, species) %>%
-    summarise(estimate = mean(estimate), LowerPI=mean(LowerPI), UpperPI=mean(UpperPI))
+    summarise(weighted_estimate = sum(estimate*weight), 
+              weighted_var = sum((weight*(estimate - weighted_estimate))^2)) %>%
+    ungroup()
+  
+  ensemble = weighted_estimates %>%
+    mutate(LowerPI= weighted_estimate - sqrt(weighted_var)*CI_sd, 
+           UpperPI= weighted_estimate + sqrt(weighted_var)*CI_sd) %>%
```

Comment:
This looks correct to me, but I had to think for a while about what "CI_sd" meant. I think it would be simpler to define something like 

`CI_offset =  qnorm((1-CI_level)/2, , sd = sqrt(weighted_var), lower.tail = FALSE)`

and then add/subtract that rather than doing part of the computation up above and part of it here.